tests/components/SpellCheck.test.tsx

```diff
@@ -38,12 +38,14 @@ describe('SpellCheck', () => {
   it('Suggestion is formatted properly', () => {
     render(<SpellCheck />);
     expect(screen.getByText('Did you mean')).toBeDefined();
-    expect(screen.getByText(mockedState.spellCheck.correctedQuery)).toBeDefined();
+    const correctedQuery = mockedState.spellCheck?.correctedQuery;
+    expect(correctedQuery && screen.getByText(correctedQuery)).toBeTruthy();
```

Comment:
Similar thing to LocationBias test, can we try updating the mockedState type to specify that spellCheck is defined and remove optional chaining? There may be more places in other test files where we can update the types too.

Reply:
I initially didn't update this file (or the others) to be like `LocationBias` because even in the `SpellCheckState`, the `correctedQuery` is optional (and same with `displayName` for facets and `displayValue` for applied query filters). So, we'd be constructing an entirely new type to force the TS compiler to realize that certain fields are defined. It seemed a little weird to be defining a bunch of new types just for the tests.

Another option could be to use a const assertion on the mocked state objects instead of giving them a `Partial<State>` type. For example, 
```ts
const mockedState = {
  spellCheck: {
    correctedQuery: 'Correction',
    enabled: false
  },
  ...
} as const;
```
I think I prefer this over creating new types in each test file. Thoughts?

Reply:
mm but that would remove the State typing for `mockedState` right? no TS suggestion or warn about accidentally add a new field to the variable

Reply:
yeah, it would only give an error if you did something incorrect and then tried passing it somewhere that expects a type of `Partial<State>`

Reply:
I do like the TS suggestions but const approach would avoid the need for optional chaining and guards. I'm okay with either way then, keep Partial<State> type or using const.